tests/ui/compiletest-self-test/only-target-has-atomic.rs

```diff
@@ -0,0 +1,9 @@
+// Check this ignored because x86_64-unknown-linux-gnu does not have 128-bit atomic operation.
+
+//@ only-x86_64-unknown-linux-gnu
+//@ needs-target-has-atomic: 128
```

Comment:
Suggestion: this *could* get invalidated in the future because the target *might* get 128-bit atomic operations in the future, but it's fine in the meantime.

Can you make this instead a `headers/tests.rs` test? E.g. https://github.com/rust-lang/rust/blob/c322cd5c5ac13b5ef16a487c889f8c148d8e3841/src/tools/compiletest/src/header/tests.rs#L705